contracts/src/v0.8/functions/dev/1_0_0/accessControl/TermsOfServiceAllowList.sol

```diff
@@ -84,24 +85,22 @@ contract TermsOfServiceAllowList is Routable, ITermsOfServiceAllowList, IAccessC
   /**
    * @inheritdoc ITermsOfServiceAllowList
    */
-  function acceptTermsOfService(address acceptor, address recipient, bytes calldata proof) external override {
+  function acceptTermsOfService(address acceptor, address recipient, bytes32 r, bytes32 s, uint8 v) external override {
     if (s_blockedSenders[recipient]) {
       revert RecipientIsBlocked();
     }
 
-    // Validate that the proof is correct and has been signed
-    if (
-      _getSigner(getEthSignedMessageHash(getMessageHash(acceptor, recipient)), proof) != s_config.proofSignerPublicKey
-    ) {
-      revert InvalidProof();
+    // Validate that the signature is correct and the correct data has been signed
+    if (ecrecover(getEthSignedMessageHash(getMessageHash(acceptor, recipient)), v, r, s) != s_config.signerPublicKey) {
```

Comment:
Maybe renaming `getEthSignedMessageHash` to `getPrefixedMessageHash` will make it more clear and resolve Steve's comment here: https://www.notion.so/chainlink/Steve-s-Functions-Solidity-Review-Notes-64447db7e6a64ad9a5f0ce503186eef6?pvs=4#1b476bcc44e14e71a7bfb55678f01e2b

Reply:
I'm going to combine them into one method in the next PR